dimod/serialization/coo.py

```diff
@@ -78,7 +78,7 @@
 from dimod import Vartype
 from dimod.binary_quadratic_model import BinaryQuadraticModel
 
-_LINE_REGEX = r'^\s*(\d+)\s+(\d+)\s+([+-]?(?:[0-9]*[.]?)[0-9]*)\s*$'
+_LINE_REGEX = r'^\s*(\d+)\s+(\d+)\s+([+-]?\d*(?:\.\d+)?)\s*$'
```

Comment:
Oh, yes, you need "triplets" below, that's why the non-capturing group. I'd recommend named groups in general, but let's leave that for another day :slightly_smiling_face: